catalog/resource_catalog.go

```diff
@@ -188,7 +194,7 @@ func ResourceCatalog() common.Resource {
 			return bindings.AddCurrentWorkspaceBindings(ctx, d, w, ci.Name, bindings.BindingsSecurableTypeCatalog)
```

Comment:
This probably needs to be moved up to the top of Update(), otherwise the WorkspaceClient used when changing provider_config.workspace_id may not have access to the isolated catalog (unless that workspace also happens to have a workspace binding separately).